kratos/input_output/logger.cpp

```diff
@@ -49,6 +49,16 @@ namespace Kratos
 		}
 	}
 
+	void Logger::Flush() {
+		auto outputs = GetOutputsInstance();
+		#pragma omp critical
+		{
+            GetDefaultOutputInstance().Flush();
+			for (auto i_output = outputs.begin(); i_output != outputs.end(); i_output++)
```

Comment:
I think it should be `++output`, otherwise afaik you create local copies
not completely sue though